File: android/app/src/main/java/jagex3/jagmisc/jagmisc.java
package jagex3.jagmisc;


public final class jagmisc {

    public static int ping0(byte arg0, byte arg1, byte arg2, byte arg3, long arg4) { return 0; }

    public static boolean init() { return false; }

    public static void Quit0() {}

    public static long nanoTime() { return System.nanoTime(); }

    public static long getTotalPhysicalMemory() { return 0L; }

    public static long getAvailablePhysicalMemory() { return 0L; }

    public static void quit() {
        Quit0();
    }

    public static int ping(byte arg0, byte arg1, byte arg2, byte arg3, long arg4) throws Throwable {
        int local6 = ping0(arg0, arg1, arg2, arg3, arg4);
        if (local6 < 0) {
            throw new Exception(String.valueOf(local6));
        }
        return local6;
    }

    public jagmisc() {
    }
}

File: src/jvmMain/kotlin/jagex3/jagmisc/jagmisc.kt
package jagex3.jagmisc

import kotlin.jvm.JvmStatic

actual object jagmisc {
    /*private*/ actual val totalPhysicalMemory: Long
        external get

    @JvmStatic
    actual val availablePhysicalMemory: Long
        external get

    /*private*/ actual external fun ping0(arg0: Byte, arg1: Byte, arg2: Byte, arg3: Byte, arg4: Long): Int

    @JvmStatic
    actual external fun init(): Boolean

    @JvmStatic
    actual external fun nanoTime(): Long

    @JvmStatic
    actual fun quit() {
        Quit0()
    }

    @JvmStatic
    @Throws(Throwable::class)
    actual fun ping(arg0: Byte, arg1: Byte, arg2: Byte, arg3: Byte, arg4: Long): Int {
        val local6 = ping0(arg0, arg1, arg2, arg3, arg4)
        if (local6 < 0) {
            throw Exception(local6.toString())
        }
        return local6
    }

    /*private*/ actual external fun Quit0()
}
